plugins/feeds/public/et_open.py

```diff
@@ -0,0 +1,138 @@
+import datetime
+import logging
+from datetime import timedelta, timezone
+from io import StringIO
+
+from idstools import rule
+
+from core import taskmanager
+from core.config.config import yeti_config
+from core.schemas import entity, indicator, task
+
+
+class ETOpen(task.FeedTask):
+    __SOURCE = (
+        "https://rules.emergingthreats.net/open/suricata-7.0.3/emerging-all.rules"
+    )
+
+    _defaults = {
+        "frequency": timedelta(days=1),
+        "name": "ETOpen",
+        "description": "ETOpen ruleset",
+        "source": __SOURCE,
+    }
+
+    def run(self):
+        response = self._make_request(self.__SOURCE, no_cache=True)
+
+        if not response:
+            return
+
+        for line in StringIO(response.text).readlines():
+            if line.startswith("#"):
+                continue
+            self.analyze(line)
+
+    def analyze(self, line: str):
+        rule_suricata = rule.parse(line)
+        if not rule_suricata:
+            return
+
+        if self.__filter_rule(rule_suricata.metadata):
+            ind_suricator = indicator.Suricata(
+                name=rule_suricata["msg"],
+                pattern=line,
+                metadata=rule_suricata.metadata,
+                diamond=indicator.DiamondModel.infrastructure,
+                sid=rule_suricata["sid"],
+            ).save()
+
+            if "cve" in ",".join(rule_suricata.metadata):
+                for ind_cve in self.__extract_cve(rule_suricata.metadata):
+                    ind_suricator.link_to(ind_cve, "affect", "ETOpen")
```

Comment:
if metadata happens to have a key that contains the string `cve`, this will break.
Given that you're already iterating, please do the check within the loop, e.g.

```python
for metadata in rule_suricata.metadata:
  if metadata.startswith("cve"): # or maybe a better check
    vulnerability = self._extract_vulnerability(metadata)
    ind_suricator.link_to(vulnerability, ...)
```